src/api/v1beta1/feature_flags.go

```diff
@@ -138,7 +138,9 @@ func (dk *DynaKube) FeatureCustomStatsdImage() string {
 	return dk.Annotations[annotationFeatureCustomStatsdImage]
 }
 
-// FeatureReadOnlyOneAgent is a feature flag that makes the operator deploy the oneagents in a readonly mode, where the csi-driver provides the volume for logs and such,
+// FeatureReadOnlyOneAgent is a feature flag to specify if the operator needs to deploy the oneagents in a readonly mode,
+// where the csi-driver would provide the volume for logs and such
+// Defaults to true
 func (dk *DynaKube) FeatureReadOnlyOneAgent() bool {
-	return dk.Annotations[AnnotationFeatureReadOnlyOneAgent] == "true"
+	return dk.Annotations[AnnotationFeatureReadOnlyOneAgent] != "false"
```

Comment:
this check is way too inclusive
since it's opt-out anyway, we should change the feature flag to `DisableReadOnlyOneAgent` similar to the other features that are opt-out